File: nest/src/databases/entities/favorites.entity.ts
import {
    CreateDateColumn,
    UpdateDateColumn,
    Entity,
    PrimaryGeneratedColumn,
    ManyToOne,
    JoinColumn,
    Column,
} from 'typeorm';
import { ArticlesEntity } from './articles.entity';
import { UsersEntity } from './users.entity';

@Entity('Favorites')
export class FavoritesEntity {
    @PrimaryGeneratedColumn()
    uid: number;

    @Column({ type: 'uuid' })
    user_id: number;

    @Column({ type: 'uuid' })
    article_id: number;

    @CreateDateColumn({
        name: 'createdAt',
    })
    created_at: Date;

    @UpdateDateColumn({
        name: 'updatedAt',
    })
    updated_at: Date;

    @ManyToOne(() => UsersEntity, users => users.uid, {
        createForeignKeyConstraints: true,
        nullable: true,
        onUpdate: 'CASCADE',
        onDelete: 'CASCADE',
    })
    @JoinColumn({
        name: 'user_id',
    })
    users: UsersEntity;

    @ManyToOne(() => ArticlesEntity, articles => articles.uid, {
        createForeignKeyConstraints: true,
        nullable: true,
        onUpdate: 'CASCADE',
        onDelete: 'CASCADE',
    })
    @JoinColumn({
        name: 'article_id',
    })
    articles: ArticlesEntity;
}

File: spring-boot/src/main/java/springboot/database/entities/FavoritesEntity.java
package springboot.database.entities;

import java.time.LocalDateTime;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

@Getter
@Setter
@Builder
@Entity(name = "Favorites")
public class FavoritesEntity {
    public FavoritesEntity() {}

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long uid;

    @ManyToOne
    @JoinColumn(name = "article_uid")
    private ArticlesEntity articlesEntity;

    @ManyToOne
    @JoinColumn(name = "user_uid")
    private UsersEntity usersEntity;

    @CreationTimestamp
    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
